compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -481,7 +516,7 @@ impl<'a, 'b, 'tcx> TypeFolder<'tcx> for AssocTypeNormalizer<'a, 'b, 'tcx> {
                         normalized_ty,
                     )
                 })
-                .unwrap_or_else(|| ty.super_fold_with(self));
+                .unwrap_or_else(|| ty.super_fold_with(self).into());
```

Comment:
don't `.into()` here, instead do the `.ty().unwrap()` 3 lines above instead of at the end of this match arm